### DISCUSSION TITLE
How to filter expand items?

### DISCUSSION THREAD

**Discussion by g1sbi** [pocketbase/pocketbase#2892] [Q&A]
I'm fetching a thread and expanding to the images collection. I only need the first image from the expansion, but I can't figure out how to filter it correctly. Is this possible to do?
```ts
const threads = await pb.collection("threads").getList(page, 20, {
    expand: "images",
    sort: "-created",
    fields: "id,title,start, end, price, category, expand",
    filter:
      "thread.expand.images.sort_order = 0 || thread.expand.images.sort_order = 1",
  });
```

(thread is obviously wrong here but it serves as an example of what I'm trying to do)

**Comment by ganigeorgiev** [comment]
No, this is not possible (unless I misunderstood your question). If you need pagination or filter for the `expand` results you can send separate request(s) for the `expand` related collection.

I have in my todo to eventually consider allowing "range" support for the `fields` parameters (see https://github.com/pocketbase/pocketbase/discussions/2783#discussioncomment-6267779) but currently this is not supported.

**Reply by g1sbi** [reply]
Understood. So I'd have to manually fetch the first image from the collection myself?

**Reply by ganigeorgiev** [reply]
Yes, if you want only the first relation item.

You can also extract the first `images` ids from the `threads` result, execute a single `images` `getFullList()` call with the extracted ids as filter and then perform the threads-image grouping client-side based on the id of each image.

Alternatively, you can also create a helper view collection, eg. `featuredThreadsImage`, and use [indirect expand](https://pocketbase.io/docs/expanding-relations/#indirect-expand) to populate the view record in the `threads` expand result (eg. `expand: featuredThreadsImage(thread)`).
